pytorch_lightning/core/optimizer.py

```diff
@@ -102,11 +102,13 @@ def _on_trainer_init(self, trainer):
                 break
 
     @classmethod
-    def to_lightning_optimizer(cls, optimizer, trainer):
-        if isinstance(optimizer, LightningOptimizer):
-            return optimizer
-        optimizer = cls(optimizer)
-        optimizer._on_trainer_init(trainer)
+    def _to_lightning_optimizer(cls, optimizer, trainer, opt_idx):
+        # apex overrides .step function and need to be wrapped on each step
```

Comment:
does it mean that `trainer.lightning_optimizers[opt_idx]` and `optimizer` are the very same optimizers just one is native the other one is PL wrapped?

Reply:
Yes, trainer.lightning_optimizers[opt_idx]._optimizer and optimizer are the same thing. trainer.lightning_optimizers[opt_idx] is the Lightning optimizer wrapping optimizer_idx optimizer.